pkg/services/internal/whatsonchain/service.go

```diff
@@ -25,9 +25,12 @@ type WhatsOnChain struct {
 	url        string
 	apiKey     string
 	logger     *slog.Logger
+
+	bsvExchangeRate   *wdk.BSVExchangeRate // TODO: possibly handle by some caching structure/redis
+	bsvUpdateInterval *time.Duration
```

Comment:
use value instead of pointer - because it's harder to maintain use it later as a pointer.